### ISSUE TITLE
Since a recent master, some images make darktable crashes when opening them on darkroom 

### ISSUE THREAD

**Issue by Nilvus** [darktable-org/darktable#14108]
**Describe the bug/issue**

I launch again, after nearly 2 weeks, darktable (and so update master), yesterday. I then discover that darktable crashes, when opening all my images of my recent filmrolls (on what I test) when I open them on darkroom.

I've check that some older images opened without any crashes. I so test to clone images that crashes, copy/paste history of good images and can open duplicates. I tested on original and same issue: no more crashes. Then I tried to remove history from lighttable for some images that crashes but again: a crash.

But images that crashes open correctly when I go back to older master I used (nearly 300 commits behind).

**To Reproduce**
1. Use XMP provided below on an image
[XMP-with-crash.zip](https://github.com/darktable-org/darktable/files/11141630/XMP-with-crash.zip)
2. Open image on darkroom
3. See loading image toast message
4. Then darktable should crash

**Expected behavior**

No crash on those images, like I can have with an older master on march.

**Which commit introduced the error**

Git bisect gives such result:

```
177b1e4f9c7c7d50b402c803b64381950fb1de1c is the first bad commit
commit 177b1e4f9c7c7d50b402c803b64381950fb1de1c
Author: Pascal Obry <pascal@obry.net>
Date:   Wed Mar 22 08:29:45 2023 +0100

    Use preset's multi_name as module label.
    
    In an attempt to give better control to the actual module label
    used (avoiding also long preset names possibly hard to read and
    ellipsize) the module label is now using the preset multi_name
    if set. If not set, the preset name is used as before.
```

So should be for you @TurboGit. As I know that sometimes git bisect is not precise, I'm sure regarding all tests, compile I've made that this is from commits merged on march 25th (same day that one had been merged). On that day, main updates are PR from @ralfbrown and @jenshannoschwalm.

**Platform**
_Please fill as much information as possible in the list given below. Please state "unknown" where you do not know the answer and remove any sections that are not applicable _

* darktable version : recent master (since at least last master of march 25th

**Comment by ralfbrown** [comment]
No crash here with either of your XMPs and the standard test image using this morning's master (4.3.0+1566).  The 03-12 does give multiple warnings from Lensfun about multiple distortion models for the lens.

**Comment by jenshannoschwalm** [comment]
OpenCL kernels cleared & recompiled?

No problem applying your xmp to any of my images ...

**Comment by Nilvus** [comment]
Thanks for your tests. I was expected that this was reproducible. I always can on such related images (and dozens other images) and nit with all master.

I can reproduce with and without opencl. And only with master since march 25th.

So is this could be issue on my database regarding some database update, regarding commit pinted by bisect?

**Comment by Nilvus** [comment]
About lensfun warnings, Isee them on images since times when I launch it from Terminal. So I doubt it's related and I never see issue on my images about that. I don't know what to do about those warnings (but never investigate those too)

**Comment by jenshannoschwalm** [comment]
Maybe this is related to some presets defined on your system and the new presets handling still has an issue.

**Comment by TurboGit** [comment]
No crash on my side either. If you can reproduce under gdb a backtrace would help. Otherwise as @jenshannoschwalm hinted maybe an issue with auto-presets. So if you can share a Zip with a RAW + xmp and the data.db it could help.

**Comment by Nilvus** [comment]
First thing, gdb backtrace (and the SIGSEV signal just before display backtrace:

```
Thread 1 "darktable" received signal SIGSEGV, Segmentation fault.
dt_presets_get_module_label (
    module_name=module_name@entry=0x55555b8214f0 "denoiseprofile", 
    params=params@entry=0x55555b686ed0, param_size=param_size@entry=412, 
    is_default_params=<optimized out>, blend_params=0x55555b66be20, 
    blend_params_size=blend_params_size@entry=420)
    at /home/nilvus/Projets/darktable/src/common/presets.c:430
430	    if(strlen(multi_name) == 0 || multi_name[0] != ' ')
(gdb) 
(gdb) bt
#0  dt_presets_get_module_label (
    module_name=module_name@entry=0x55555b8214f0 "denoiseprofile", 
    params=params@entry=0x55555b686ed0, param_size=param_size@entry=412, 
    is_default_params=<optimized out>, blend_params=0x55555b66be20, 
    blend_params_size=blend_params_size@entry=420)
    at /home/nilvus/Projets/darktable/src/common/presets.c:430
#1  0x00007ffff7a516ae in _iop_update_label (data=0x55555b821310)
    at /home/nilvus/Projets/darktable/src/develop/imageop.c:1904
#2  0x00007ffff743d8fa in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff743cdbf in g_main_context_dispatch ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff743d198 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff743d44f in g_main_loop_run ()
   from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#6  0x00007ffff6c08435 in gtk_main () from /lib/x86_64-linux-gnu/libgtk-3.so.0
#7  0x00007ffff7af5fc9 in dt_gui_gtk_run (gui=<optimized out>)
    at /home/nilvus/Projets/darktable/src/gui/gtk.c:1284
#8  0x00005555555550a0 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/nilvus/Projets/darktable/src/main.c:94
```
I have now understand the issue. I have on denoise profile an auto preset (since darktable 3.0) called "Réduction du bruit de profil 3.0. I have checked and all images affected have that label on denoiseprofile. Images that do not crash do not have that label. I have tried on image without label (so no crash). As soon as I select that auto-preset, darktable crash. New label name on auto-presets do not like such notation with x.x like 3.0 there. This happens only on that preset.

I have renamed that preset via darktable tab on preferences and override that module on some images with crash and I can now open those images.

I can fix that so easily and quickly on affected images but could be good to deal with such names on darktable for other users. Thanks again for you help.

**Comment by Nilvus** [comment]
I also see that on that module (noise reduction (profil)), I can't anymore have label of my presets displayed beside module name. I always have "1" instead (even if I reset or start from a new image on scratch) on history and on module title. I even delete my presets on that module and recreate new ones with new labels.  On some other modules tested (exposure, filmic, sharpen & diffuse), this works as expected. This seems to only affect that module. I have some images with also 2 instances and both have so "1" even if presets used are not the same. That so seems to make darktable consider those 2 instances as same module and changes, as I can see on history. So no more 2 real instances.

Maybe all that is related (and not the 3.0 writing) as label presets seems to not be considered correctly on that module.

**Comment by Nilvus** [comment]
Another thing: I just did a new test. On relevant PR #13998, had been added the option to automatically update the module name in the preferences (in darkroom - modules). If I uncheck it, I see that the "1" is only set on my automatic preset. In short, I have 2 presets for this module: one activated automatically and another on the Y0U0V0 mode that I add manually in 2nd instance for some images. If I reactivate this auto update, the "1" appears on both instances. So there is already a bug on this point. I have the feeling that all that is related and causes the crash.
